src/reactviews/pages/SchemaDesigner/schemaDesigner.tsx

```diff
@@ -3,102 +3,411 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { useContext, useEffect } from "react";
+import { useContext, useEffect, useRef, useState } from "react";
 import { SchemaDesignerContext } from "./schemaDesignerStateProvider";
 import * as azdataGraph from "azdataGraph";
 import "azdataGraph/dist/index.css";
 import "azdataGraph/src/css/common.css";
 import "azdataGraph/src/css/explorer.css";
 import "./schemaDesigner.css";
-import * as schemaDesignerIcons from "./schemaDesignerIcons";
+import { config, getSchemaDesignerColors } from "./schemaDesignerConfig";
+import { mxCell } from "mxgraph";
+import { SchemaDesignerTableEditor } from "./schemaDesignerEntityEditor";
+import { ITable } from "../../../sharedInterfaces/schemaDesigner";
+import {
+    Menu,
+    MenuItem,
+    MenuList,
+    MenuPopover,
+    PositioningImperativeRef,
+} from "@fluentui/react-components";
+import * as htmlToImage from "html-to-image";
 
-const connectorIcon = require("./icons/connector.svg");
+// Set the global mxLoadResources to false to prevent mxgraph from loading resources
+window["mxLoadResources"] = false;
 
 export const SchemaDesigner = () => {
     const context = useContext(SchemaDesignerContext);
-    window["mxLoadResources"] = false;
     if (!context) {
         return undefined;
     }
 
+    const [displayEditor, setDisplayEditor] = useState(false);
+
+    const [table, setTable] = useState<ITable | undefined>(undefined);
+
+    const [schema, setSchema] = useState<azdataGraph.ISchema>(context.schema);
+
+    const graphContainerRef = useRef<HTMLDivElement | null>(null);
+    const editorDivRef = useRef<HTMLDivElement | null>(null);
+
+    const [schemaDesigner, setSchemaDesigner] = useState<
+        azdataGraph.SchemaDesigner | undefined
+    >(undefined);
+
+    const [exportAsMenuOpen, setExportAsMenuOpen] = useState(false);
+    const exportPositioningRef = useRef<PositioningImperativeRef>(null);
+
     useEffect(() => {
+        context.extensionRpc.subscribe(
+            "schemaDesigner",
+            "onDidChangeTheme",
+            (_params) => {
+                if (schemaDesigner) {
+                    schemaDesigner.applyColors(getSchemaDesignerColors());
+                }
+            },
+        );
+    }, [schemaDesigner]);
+
+    useEffect(() => {
+        const editorDiv = editorDivRef.current;
+        function updateEditorPosition(x: number, y: number, scale: number) {
+            if (!editorDiv) {
+                return;
+            }
+            editorDiv.style.transform = `scale(${scale})`;
+            editorDiv.style.top = `${y}px`;
+            editorDiv.style.left = `${x}px`;
+        }
         function createGraph() {

```

Comment:
this is a long function to stick in an effect; can you shift it out to a helper function and add comment?